doc/extensions/best-practices.rst

```diff
@@ -37,17 +39,61 @@ Implement each plugin class in a separate Python module
 This keeps CKAN's plugin loading order simple, see :ref:`ckan.plugins`.
 
 
-.. _extension config setting names best practice:
+.. _avoid name clashes:
+
+------------------
+Avoid name clashes
+------------------
+Many of the names you pick for your identifiers and files must be unique in
+relation to the names used by core CKAN and other extensions. To avoid
+conflicts you should prefix any public name that your extension introduces with
+the name of your extension. For example:
+
+* The names of *configuration settings* introduced by your extension should
+  have the form ``ckan.my_extension.my_config_setting``.
```

Comment:
The two patterns I've seen for configuration options are
```
ckanext.my_extension.my_config_setting = ...
```

and

```
my_extension.my_config_setting = ...
```

I choose the latter for my extensions because I find the configuration options are really long with the former.
Anything starting with `ckan.` should belong to ckan core.